src/mlib/str.h

```diff
@@ -825,13 +828,15 @@ mstr_win32_narrow (const wchar_t *wstring)
       return (mstr_narrow_result){.string = MSTR_NULL,
                                   .error = GetLastError ()};
    }
-   mstr_mut ret = mstr_new ((size_t) length);
+   // Allocate a new string, not including the null terminator
+   mstr_mut ret = mstr_new ((size_t) (length - 1));
    int got_len = WideCharToMultiByte (CP_UTF8,
-                                      WC_ERR_INVALID_CHARS,
+                                      wcflags,
                                       wstring,
-                                      -1,
+                                      length,
```

Comment:
```suggestion
                                      wcslen (wstring),
```

IIUC `length` is the required size of `ret.data`. This argument, `cchWideChar` is:

> Size, in characters, of the string indicated by lpWideCharStr.

I think if the `wstring` contains a character that encodes to multi-byte UTF-8, this may be an incorrect length for `wstring`.

If wstring is null terminated.

Reply:
I'll add a proper test using some utf16 strings to make sure.